### ISSUE TITLE
Command "l" (lowercase L, render text report) removed in BrewPi 0.5.5 firmware

### ISSUE THREAD

**Issue by NamelessCoder** [thorrak/fermentrack#278]
Hi @thorrak,

Thanks for a neat little app. After some struggle (caused mostly by BrewPi itself) I managed to get a setup running, but because I didn't think twice about upgrading firmwares, I ended up running with BrewPi 0.5.5.

This would be fine I guess, except it seems that this version removed the `l` command which is heavily used by Fermentrack to fetch the emulated LCD output:

https://github.com/BrewPi/firmware/commit/39d0b404fbb8c37593356a07c1597f47cc6cd802#diff-483635118eb441b25b12986056941a7bL339

Which has resulted in my stderr log from Fermentrack being full of:

```
 May 20 2018 23:06:39   Controller debug message: WARNING 1: Invalid command received by Arduino: l
 May 20 2018 23:06:44   Controller debug message: WARNING 1: Invalid command received by Arduino: l
 May 20 2018 23:06:49   Controller debug message: WARNING 1: Invalid command received by Arduino: l
 May 20 2018 23:06:54   Controller debug message: WARNING 1: Invalid command received by Arduino: l
```

You can imagine with this frequency the log fills up rather quickly and it can be hard to detect actual errors.

IMHO the emulated LCD readout is not really necessary as the meters convey the same information. Would be fine to remove this feature imho (and that might also reduce the number of requests made to the device which seem to be quite a lot?)

Thanks in advance for considering!

**EDIT: Related to this, to enable some "live" meter outputs before starting logging, perhaps the main sensors could be queried and updated instead of showing the LCD emu?** First time users may find it confusing that no output happens until logging starts - I know I did. It made me wonder many times if perhaps I hadn't configured correctly.

**Comment by thorrak** [comment]
Are you using Fermentrack with an OEM BrewPi?

**Comment by NamelessCoder** [comment]
I'm using the one sold on https://store.brewpi.com/featured/brewpi-spark-3, I used their official tool for upgrading the firmware which is what gave me 0.5.5 yesterday - and I'm connecting to it over WiFi from a Raspberry 3 B+ with Raspbian.

**Comment by NamelessCoder** [comment]
> That’s wild - you may be the first person using an OEM BrewPi with Fermentrack

Great, in that case, I've got the time and motivation to help you perfect the integration with Fermentrack. My Python skills aren't exactly awesome (but I get by) - and I can most definitely help you with both debugging and if you want, access to the raspberry that runs a live setup with this component!

> Not to derail your request, but out of curiosity what issues did you encounter when attaching the BrewPi?

I'm actually fighting one of those issues right now and was going to report that as a separate issue report - most importantly, the PIN outputs do not seem to be mapped to the right connections. I am experiencing two separate issues which I think are directly connected:

1. The "actuator" outputs don't seem to be mapped to the PIN outputs. I've been through all of the PIN outputs now, connecting my SSR to each and allowing the software to enter cooling state - none of them have had an effect. I'm going to make a second check to confirm that brewpi-www can speak to it and that I did my wiring correctly.
2. I used this device to connect to the SSR with one-wire: https://store.brewpi.com/temperature-control/solid-state-relays-ssr/brewpi-onewire-ssr-expansion-board. When it is connected, an additional two unknown devices appear in the list of connected devices, but they have no settings and cannot be assigned. I noticed something similar in brewpi-www where the SSR appears as two different devices (I guess for the A output and forwarding through B...) and only one of them toggles the SSR. These devices no longer appear since I switched to directly connecting the SSR to one of the "actuator" outputs.

I'll of course update here if I locate the problem when I go through it again.

> To your point, however - What I’m leaning towards doing to correct for this is having BrewPi-script “emulate” the LCD output by just rebuilding it from the meter outputs/control settings cache as you noted. Too much of Fermentrack’s UI is built around the LCD to allow for eliminating it at this point without things feeling weird.

That doesn't sound unreasonable of course. Due to the circumstances I can't even tell what the LCD readout can do other than showing this controller reply and the "starting script" message?

If it only does those two things however, maybe it's cleaner UI if there were a "Controller state" readout as alternative view in one of the dash panels that could show either "Starting script...", the most recent error message, or the cooling/heating/idle state - as appropriate. Such a readout should then be the default view instead of for example the log interval/configure device one (and perhaps "Configure device" link could be sacrificed since that's already in the top menu).

On a side note I'm doing a bit of hacking to improve the UI of the dashboard in particular, for a more compact view on desktop and a properly scaling view on mobile devices. If that turns out well I'll of course prepare a pull request to suggest the changes.

...will return with more info after hardware debugging.

![screen shot 2018-05-21 at 14 51 11](https://user-images.githubusercontent.com/2228052/40308411-79e99e06-5d06-11e8-9faa-22c6acd06ac5.png)
![screen shot 2018-05-21 at 14 50 54](https://user-images.githubusercontent.com/2228052/40308412-7a036098-5d06-11e8-9067-c5631fba5336.png)
![screen shot 2018-05-21 at 14 50 39](https://user-images.githubusercontent.com/2228052/40308413-7a1bf7de-5d06-11e8-9269-0b1e07db838f.png)

**Comment by NamelessCoder** [comment]
I wrote:

> The "actuator" outputs don't seem to be mapped to the PIN outputs.

This looks like a brainfart on my behalf. "Someone" didn't connect things correctly. Now I just have to identify the PIN connected to the output I need to use to fit the brewpi in the mount I built... so strike this issue!

The other one, that the SSR expansion board appears as two unknown devices, would be great to solve since it's a bit more flexible in terms of wiring due to the extra temp sensor ports that lets me hide stuff in the fridge compressor compartment with the SSR.
